packages/aws-rfdk/lib/deadline/scripts/python/configure_identity_registration_settings.py

```diff
@@ -0,0 +1,607 @@
+                desired_rfdk_setting = LoadBalancerIdentityRegistrationSetting(**desired_rfdk_setting_fields)
+
+                if setting_differs(prior_rfdk_setting, desired_rfdk_setting):
+                    update_setting(dl_secrets, desired_rfdk_setting)
+                else:
+                    print(f'setting "{setting_name}" exists and is up-to-date, skipping')
+            else:
+                create_setting(dl_secrets, desired_rfdk_setting)
+
+
+def setting_differs(
+    setting_a: LoadBalancerIdentityRegistrationSetting,
+    setting_b: LoadBalancerIdentityRegistrationSetting,
+) -> bool:
+    return setting_a != setting_b
+
+
+def update_setting(
+    dl_secrets: DeadlineSecretsCommandClient,
+    setting: LoadBalancerIdentityRegistrationSetting,
+) -> None:
+    print(json.dumps(
+        {
+            "action": "update",
+            "setting": setting._asdict(),
+        },
+        indent=2,
+    ))
+    print(dl_secrets.run_str(
+        "UpdateLoadBalancerIdentityRegistrationSetting",
+        setting.settings_id,
+        setting.settings_name,
+        setting.connection_ip_filter_type,
+        setting.connection_ip_filter_value,
+        setting.source_ip_filter_type,
+        setting.source_ip_filter_value,
+        setting.default_role,
+        setting.default_status,
+        str(setting.is_enabled),
+    ))
+
+
+def create_setting(
+    dl_secrets: DeadlineSecretsCommandClient,
+    setting: LoadBalancerIdentityRegistrationSetting,
+) -> None:
+    print(json.dumps(
+        {
+            "action": "create",
+            "setting": setting._asdict(),
+        },
+        indent=2,
+    ))
+    print(dl_secrets.run_str(
+        "CreateLoadBalancerIdentityRegistrationSetting",
+        setting.settings_name,
+        setting.connection_ip_filter_type,
+        setting.connection_ip_filter_value,
+        setting.source_ip_filter_type,
+        setting.source_ip_filter_value,
+        setting.default_role,
+        setting.default_status,
+        str(setting.is_enabled),
+    ))
+
+
+def apply_registration_settings(config):
+    connection_subnets = config.connection_subnet # type: List[str]
+    credentials = json.loads(fetch_secret(config.credentials))
+    source_subnets = config.source_subnet # type: List[SourceSubnet]
+
+    # Get the CIDR ranges of all subnets involved in the desired registration setting rules
+    all_subnets = set(connection_subnets).union(
+        source_subnet.subnet_id for source_subnet in source_subnets
+    )
+    subnet_to_cidr = get_subnet_cidrs(config.region, all_subnets)
+    print('subnet_to_cidr = ' + json.dumps(subnet_to_cidr, indent=4))
+
+    dl_secrets = DeadlineSecretsCommandClient(credentials['username'], credentials['password'])
+
+    prior_rfdk_settings = get_rfdk_registration_settings(dl_secrets)
+
+    delete_removed_settings(dl_secrets, prior_rfdk_settings, connection_subnets, source_subnets)
```

Comment:
Am I understand right that settings that created manually will be removed?

Reply:
The code uses a RFDK-specific convention for naming the identity registration settings. The settings have a name convention of:

```
RfdkSubnet|<CONNECTING_SUBNET_ID>|<SOURCE_SUBNET_ID>
```

This code will only remove settings that:

1.  match the above naming convention
2.  do not match the connecting/source subnets supplied in the command-line arguments 

So the only scenario where it would delete a manually created identity registration setting is if a user created one that matches the RFDK naming convention.